### DISCUSSION TITLE
Do we need indexes in lines?

### DISCUSSION THREAD

**Discussion by samlown** [invopop/gobl#81] [Polls]
Suddenly had a thought, maybe the "Index" or `i` field in Invoice lines is not very useful:

https://github.com/invopop/gobl/blob/b8eb2ee7adee7c545c49dedba5299172fc7c431a/bill/line.go#L15-L16

It's auto calculated as the index of the row in the array, plus one, so in reality adds very little value, and potentially adds noise.

**Comment by cavalle** [comment]
I suppose we could live without it, but I do see some value on having an attribute that indicates explicitly and 
unequivocally the index of an invoice line. It also serves as an identifier of the line. And it's not an uncommon field in the invoicing domain (for example, DIAN's UBL expects consecutive IDs for each of their `InvoiceLine`s). 

I don't know if this field is causing trouble somewhere else but, based on my knowledge, my vote goes to keeping it.
